### ISSUE TITLE
APN中USHCN的batchsize问题，以及跑出来的MIMIC-Ⅲ的数据和论文相比差了好多

### ISSUE THREAD

**Issue by ggal63475-debug** [Ladbaby/PyOmniTS#12]
1.原APN的USHCN的batchsize是32，您的配置是16，我改成了32，跑出来之后和原实验差了一些，请问这是为什么？
下面是用您仓库的APN 并且把batchsize改为32跑出来的结果：
Testing:  25%|██████████▌                               | 1/4 [00:00<00:01,  2.42it/s]2026-04-14 11:14:48 WARNING Exp_Main.test(): Batch No.3 out of [0~3] has an actual batch_size=16, which is not the same as --batch_size 32
2026-04-14 11:14:48 INFO Metrics:                                                     
{
    "MAE": 0.26790595054626465,
    "MSE": 0.17212365567684174
}
2026-04-14 11:14:48 INFO Loading model checkpoint from storage/results/USHCN/USHCN/APN/APN/150_3/2026_0414_1110/iter1/pytorch_model.bin
2026-04-14 11:14:48 INFO Testing results will be saved under storage/results/USHCN/USHCN/APN/APN/150_3/2026_0414_1110/iter1/eval_2026_0414_1114
Testing:  25%|██████████▌                               | 1/4 [00:00<00:01,  2.33it/s]2026-04-14 11:14:48 WARNING Exp_Main.test(): Batch No.3 out of [0~3] has an actual batch_size=16, which is not the same as --batch_size 32
2026-04-14 11:14:49 INFO Metrics:                                                     
{
    "MAE": 0.276785284280777,
    "MSE": 0.18570446968078613
}
2026-04-14 11:14:49 INFO Loading model checkpoint from storage/results/USHCN/USHCN/APN/APN/150_3/2026_0414_1110/iter2/pytorch_model.bin
2026-04-14 11:14:49 INFO Testing results will be saved under storage/results/USHCN/USHCN/APN/APN/150_3/2026_0414_1110/iter2/eval_2026_0414_1114
Testing:  25%|██████████▌                               | 1/4 [00:00<00:01,  2.50it/s]2026-04-14 11:14:49 WARNING Exp_Main.test(): Batch No.3 out of [0~3] has an actual batch_size=16, which is not the same as --batch_size 32
2026-04-14 11:14:49 INFO Metrics:                                                     
{
    "MAE": 0.28045496344566345,
    "MSE": 0.17977508902549744
}
2026-04-14 11:14:49 INFO Loading model checkpoint from storage/results/USHCN/USHCN/APN/APN/150_3/2026_0414_1110/iter3/pytorch_model.bin
2026-04-14 11:14:49 INFO Testing results will be saved under storage/results/USHCN/USHCN/APN/APN/150_3/2026_0414_1110/iter3/eval_2026_0414_1114
Testing:  25%|██████████▌                               | 1/4 [00:00<00:01,  2.44it/s]2026-04-14 11:14:50 WARNING Exp_Main.test(): Batch No.3 out of [0~3] has an actual batch_size=16, which is not the same as --batch_size 32
2026-04-14 11:14:50 INFO Metrics:                                                     
{
    "MAE": 0.3566663861274719,
    "MSE": 0.30291885137557983
}
2026-04-14 11:14:50 INFO Loading model checkpoint from storage/results/USHCN/USHCN/APN/APN/150_3/2026_0414_1110/iter4/pytorch_model.bin
2026-04-14 11:14:50 INFO Testing results will be saved under storage/results/USHCN/USHCN/APN/APN/150_3/2026_0414_1110/iter4/eval_2026_0414_1114
Testing:  25%|██████████▌                               | 1/4 [00:00<00:01,  2.73it/s]2026-04-14 11:14:50 WARNING Exp_Main.test(): Batch No.3 out of [0~3] has an actual batch_size=16, which is not the same as --batch_size 32
2026-04-14 11:14:50 INFO Metrics:                                                     
{
    "MAE": 0.2805975079536438,
    "MSE": 0.2319469302892685
}
2.再就是我没动任何的配置，用APN跑MIMIC-Ⅲ，跑出来的数据和论文给出的数据相比差了好多

**Comment by Ladbaby** [comment]
- 前情提要：APN 基于 PyOmniTS v1.0.0
- 一句话概括：PyOmniTS v2.0.0 中，MIMIC-III, MIMIC-IV, P12, USHCN 的训练/验证/测试集划分和 v1.0.0 时有所不同。涉及到的改动请参考这个 [commit](https://github.com/Ladbaby/PyOmniTS/commit/e32d234336e41aa80b224d364ca886d2d5aff359)。如果想在 PyOmniTS v2.0.0 的基准下让 APN 跑出更好的性能，请尝试用 `--sweep 1` 重新搜索超参数，此内容在 APN 首次加入的 [v1.3.2](https://github.com/Ladbaby/PyOmniTS/releases/tag/v1.3.2) 版本中有所说明。
- 为什么做这样的改动：由于 v1.0.0 中，shuffle 时未提供 random seed，我们担心随着 sklearn 版本的迭代，shuffle 的结果会发生变化，导致数据集在 `__getitem__` 时提供的 sampleID 变得不准确，故尝试在 tsdm 中禁用了 shuffle（由于 USHCN 数据集在改动前后性能差异过大，我们后续进一步把 USHCN 改成了提供固定 random seed 的方式）。作为补偿，我们为所有 PyOmniTS 中已经存在的模型都重新在新的基准上搜索了超参数（APN此时还未加入到 PyOmniTS），使得几乎所有模型在 v2.0.0 上都比 v1.0.0 上有更好的性能表现。

PS: PyOmniTS 对于 5 个不规则数据集的 shuffle 方法，是 5 轮给 numpy 和 torch 设置 5 个不同的 random seed，而并非选择 5 种不同的训练/验证/测试集划分。因为假如用后者的话，不规则时序数据集由于样本相对较少且有较大的样本差异，最终的标准差会大的离谱。

**Comment by ggal63475-debug** [comment]
谢谢作者给我这个小白如此细心的回复！再次感谢！！对我帮助特别大
